### DISCUSSION TITLE
ImagickDriver does not support `save` of extensionless file

### DISCUSSION THREAD

**Discussion by Tofandel** [spatie/image#236] [General]
Saving images without extension was possible in v2

Now running `$image->save('jpg:some_file_without_ext')` which Imagick technically supports will throw an `UnsupportedImageFormat('')`

Because of the hardcoded check on extension

```php
$extension = pathinfo($path, PATHINFO_EXTENSION);

        if (! in_array(strtoupper($extension), Imagick::queryFormats('*'))) {
            throw UnsupportedImageFormat::make($extension);
        }
```

**Comment by timvandijck** [comment]
This is correct, to make things convenient we decided to derive the file formaths from the file path.

We would probably accept a PR that adds an optional extension param to the save method as a fallback.

**Reply by laserhybiz** [reply]
This would be a breaking change since we need to modify the interface

**Reply by Tofandel** [reply]
It doesn't necessarily need to be, the extension can be added directly in the string like in the description `$image->save('jpg:some_file_without_ext')` it's how Imagick consumes it anyway

**Comment by laserhybiz** [comment]
I can't use this package anymore to modify uploaded images, is there a way to include the extension in uploaded images?
